include/deal.II/dofs/dof_tools.h

```diff
@@ -1437,6 +1437,39 @@ namespace DoFTools
                                          const std::set<types::boundary_id> &boundary_ids = std::set<types::boundary_id>());
 
   /**
+   * Extract all degrees of freedom (DoFs) that have support only within cells,
+   * for which @p predicate is <code>true</code>. The result is returned as
+   * an IndexSet.
+   *
+   * Consider the following FE space where predicate returns <code>true</code>
+   * for all cells on the left side:
+   *
+   * @image html extract_dofs_with_support_on.png
+   *
+   * This functions will return the union of all DoFs on those cells minus
+   * DoF 11, 13, 2 and 0; the result will be <code>[9,10], 12, [14,38]</code>.
```

Comment:
By the way, dof index 4 is also to the right of the line -- I think you want to add it to the list here